Missing import

I project that uses `megaparsec` and `megaparsec-9.8.0` (released in the last day or two) is failing to build:
```
Building library for megaparsec-9.8.0...
[ 1 of 19] Compiling Text.Megaparsec.Pos ( Text/Megaparsec/Pos.hs, dist/build/Text/Megaparsec/Pos.o, dist/build/Text/Megaparsec/Pos.dyn_o )
[ 2 of 19] Compiling Text.Megaparsec.Unicode ( Text/Megaparsec/Unicode.hs, dist/build/Text/Megaparsec/Unicode.o, dist/build/Text/Megaparsec/Unicode.dyn_o )
[ 3 of 19] Compiling Text.Megaparsec.Error[boot] ( Text/Megaparsec/Error.hs-boot, dist/build/Text/Megaparsec/Error.o-boot, dist/build/Text/Megaparsec/Error.dyn_o-boot )
[ 4 of 19] Compiling Text.Megaparsec.State ( Text/Megaparsec/State.hs, dist/build/Text/Megaparsec/State.o, dist/build/Text/Megaparsec/State.dyn_o )
[ 5 of 19] Compiling Text.Megaparsec.Stream ( Text/Megaparsec/Stream.hs, dist/build/Text/Megaparsec/Stream.o, dist/build/Text/Megaparsec/Stream.dyn_o )

Text/Megaparsec/Stream.hs:517:26: error: [GHC-88464]
    Variable not in scope:
      foldl' :: (b -> Char -> b) -> b -> [Char] -> b
    Suggested fixes:
      • Perhaps use one of these:
          ‘B.foldl'’ (imported from Data.ByteString),
          ‘BL.foldl'’ (imported from Data.ByteString.Lazy),
          ‘T.foldl'’ (imported from Data.Text)
      • Perhaps you want to add ‘foldl'’ to the import list
        in the import of ‘Data.Foldable’
        (Text/Megaparsec/Stream.hs:43:1-29).
    |
517 |     reachOffset' splitAt foldl' id id ('\n', '\t') charInc o pst
    |                          ^^^^^^
```
I suspect that this issue is GHC version dependent (or at least `base` library version dependent).

Would be happy to provide a PR for this and include a CI rejig that includes all recent versions of GHC.

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -13,3 +13,4 @@ jobs:
     uses: mrkkrp/haskell-ci/.github/workflows/haskell-ci.yml@master
     with:
       additional-packages: 'megaparsec-tests'
+      ghc-versions: '["9.6.7", "9.8.4", "9.10.3", "9.12.4", "9.14.1"]'

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,8 @@
 * Fixed the regression introduced by the fix for the [issue
   572](https://github.com/mrkkrp/megaparsec/issues/572) which caused the
   position marker `^` to be missing in certain cases.
+* This release officially supports GHC 9.6. This is the oldest GHC version
+  we support at this time.
 
 ## Megaparsec 9.8.0
 

--- a/Text/Megaparsec/Byte/Lexer.hs
+++ b/Text/Megaparsec/Byte/Lexer.hs
@@ -41,6 +41,7 @@ where
 
 import Control.Applicative
 import Data.Functor (void)
+import qualified Data.List
 import Data.Proxy
 import Data.Scientific (Scientific)
 import qualified Data.Scientific as Sci
@@ -121,7 +122,7 @@ decimal_ ::
   m a
 decimal_ = mkNum <$> takeWhile1P (Just "digit") isDigit
   where
-    mkNum = fromInteger . foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
+    mkNum = fromInteger . Data.List.foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
     step a w = a * 10 + fromIntegral (w - 48)
 {-# INLINE decimal_ #-}
 
@@ -144,7 +145,7 @@ binary =
     <$> takeWhile1P Nothing isBinDigit
     <?> "binary integer"
   where
-    mkNum = fromInteger . foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
+    mkNum = fromInteger . Data.List.foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
     step a w = a * 2 + fromIntegral (w - 48)
     isBinDigit w = w == 48 || w == 49
 {-# INLINEABLE binary #-}
@@ -169,7 +170,7 @@ octal =
     <$> takeWhile1P Nothing isOctDigit
     <?> "octal integer"
   where
-    mkNum = fromInteger . foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
+    mkNum = fromInteger . Data.List.foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
     step a w = a * 8 + fromIntegral (w - 48)
     isOctDigit w = w - 48 < 8
 {-# INLINEABLE octal #-}
@@ -194,7 +195,7 @@ hexadecimal =
     <$> takeWhile1P Nothing isHexDigit
     <?> "hexadecimal integer"
   where
-    mkNum = fromInteger . foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
+    mkNum = fromInteger . Data.List.foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
     step a w
       | w >= 48 && w <= 57 = a * 16 + fromIntegral (w - 48)
       | w >= 97 = a * 16 + fromIntegral (w - 87)
@@ -255,7 +256,7 @@ dotDecimal_ ::
   m SP
 dotDecimal_ pxy c' = do
   void (B.char 46)
-  let mkNum = foldl' step (SP c' 0) . chunkToTokens pxy
+  let mkNum = Data.List.foldl' step (SP c' 0) . chunkToTokens pxy
       step (SP a e') w =
         SP
           (a * 10 + fromIntegral (w - 48))

--- a/Text/Megaparsec/Char/Lexer.hs
+++ b/Text/Megaparsec/Char/Lexer.hs
@@ -69,6 +69,7 @@ where
 import Control.Applicative
 import Control.Monad (void)
 import qualified Data.Char as Char
+import qualified Data.List
 import Data.List.NonEmpty (NonEmpty (..))
 import Data.Maybe (fromMaybe, isJust, listToMaybe)
 import Data.Proxy
@@ -371,7 +372,7 @@ decimal_ ::
   m a
 decimal_ = mkNum <$> takeWhile1P (Just "digit") Char.isDigit
   where
-    mkNum = fromInteger . foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
+    mkNum = fromInteger . Data.List.foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
     step a c = a * 10 + fromIntegral (Char.digitToInt c)
 {-# INLINE decimal_ #-}
 
@@ -394,7 +395,7 @@ binary =
     <$> takeWhile1P Nothing isBinDigit
     <?> "binary integer"
   where
-    mkNum = fromInteger . foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
+    mkNum = fromInteger . Data.List.foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
     step a c = a * 2 + fromIntegral (Char.digitToInt c)
     isBinDigit x = x == '0' || x == '1'
 {-# INLINEABLE binary #-}
@@ -422,7 +423,7 @@ octal =
     <$> takeWhile1P Nothing Char.isOctDigit
     <?> "octal integer"
   where
-    mkNum = fromInteger . foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
+    mkNum = fromInteger . Data.List.foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
     step a c = a * 8 + fromIntegral (Char.digitToInt c)
 {-# INLINEABLE octal #-}
 
@@ -449,7 +450,7 @@ hexadecimal =
     <$> takeWhile1P Nothing Char.isHexDigit
     <?> "hexadecimal integer"
   where
-    mkNum = fromInteger . foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
+    mkNum = fromInteger . Data.List.foldl' step 0 . chunkToTokens (Proxy :: Proxy s)
     step a c = a * 16 + fromIntegral (Char.digitToInt c)
 {-# INLINEABLE hexadecimal #-}
 
@@ -509,7 +510,7 @@ dotDecimal_ ::
   m SP
 dotDecimal_ pxy c' = do
   void (C.char '.')
-  let mkNum = foldl' step (SP c' 0) . chunkToTokens pxy
+  let mkNum = Data.List.foldl' step (SP c' 0) . chunkToTokens pxy
       step (SP a e') c =
         SP
           (a * 10 + fromIntegral (Char.digitToInt c))

--- a/Text/Megaparsec/Stream.hs
+++ b/Text/Megaparsec/Stream.hs
@@ -42,6 +42,7 @@ import qualified Data.ByteString.Lazy.Char8 as BL8
 import Data.Char (chr)
 import Data.Foldable (toList)
 import Data.Kind (Type)
+import qualified Data.List
 import Data.List.NonEmpty (NonEmpty (..))
 import qualified Data.List.NonEmpty as NE
 import Data.Maybe (fromMaybe)
@@ -514,9 +515,9 @@ class (Stream s) => TraversableStream s where
 instance TraversableStream String where
   -- NOTE Do not eta-reduce these (breaks inlining)
   reachOffset o pst =
-    reachOffset' splitAt foldl' id id ('\n', '\t') charInc o pst
+    reachOffset' splitAt Data.List.foldl' id id ('\n', '\t') charInc o pst
   reachOffsetNoLine o pst =
-    reachOffsetNoLine' splitAt foldl' ('\n', '\t') charInc o pst
+    reachOffsetNoLine' splitAt Data.List.foldl' ('\n', '\t') charInc o pst
 
 instance TraversableStream B.ByteString where
   -- NOTE Do not eta-reduce these (breaks inlining)

--- a/megaparsec-tests/megaparsec-tests.cabal
+++ b/megaparsec-tests/megaparsec-tests.cabal
@@ -5,7 +5,9 @@ license:         BSD-2-Clause
 license-file:    LICENSE.md
 maintainer:      Mark Karpov <markkarpov92@gmail.com>
 author:          Megaparsec contributors
-tested-with:     ghc ==9.10.3 ghc ==9.12.4 ghc ==9.14.1
+tested-with:
+    ghc ==9.6.7 ghc ==9.8.4 ghc ==9.10.3 ghc ==9.12.4 ghc ==9.14.1
+
 homepage:        https://github.com/mrkkrp/megaparsec
 bug-reports:     https://github.com/mrkkrp/megaparsec/issues
 synopsis:        Test utilities and the test suite of Megaparsec
@@ -25,7 +27,7 @@ library
     default-language: Haskell2010
     build-depends:
         QuickCheck >=2.17 && <2.19,
-        base >=4.20 && <5,
+        base >=4.18 && <5,
         bytestring >=0.2 && <0.13,
         containers >=0.5 && <0.9,
         hspec >=2 && <3,
@@ -64,7 +66,7 @@ test-suite tests
     default-language:   Haskell2010
     build-depends:
         QuickCheck >=2.17 && <2.19,
-        base >=4.20 && <5,
+        base >=4.18 && <5,
         bytestring >=0.2 && <0.13,
         case-insensitive >=1.2 && <1.3,
         containers >=0.5 && <0.9,

--- a/megaparsec-tests/tests/Text/Megaparsec/StreamSpec.hs
+++ b/megaparsec-tests/tests/Text/Megaparsec/StreamSpec.hs
@@ -8,6 +8,7 @@ import Control.Monad
 import qualified Data.ByteString as B
 import qualified Data.ByteString.Lazy as BL
 import Data.Char (chr, isControl, isLetter, isSpace)
+import qualified Data.List
 import Data.List.NonEmpty (NonEmpty (..))
 import qualified Data.List.NonEmpty as NE
 import Data.Proxy
@@ -519,7 +520,7 @@ describeReachOffset Proxy =
       property $ \os' (NonNegative d) s -> do
         let os = getNonNegative <$> os'
             s' :: PosState String
-            s' = foldl' f s os
+            s' = Data.List.foldl' f s os
             o' = case os of
               [] -> d
               xs -> maximum xs + d
@@ -576,7 +577,7 @@ describeReachOffsetNoLine Proxy =
       property $ \os' (NonNegative d) s -> do
         let os = getNonNegative <$> os'
             s' :: PosState String
-            s' = foldl' f s os
+            s' = Data.List.foldl' f s os
             o' = case os of
               [] -> d
               xs -> maximum xs + d

--- a/megaparsec.cabal
+++ b/megaparsec.cabal
@@ -9,7 +9,9 @@ author:
     Paolo Martini <paolo@nemail.it>,
     Daan Leijen <daan@microsoft.com>
 
-tested-with:     ghc ==9.10.3 ghc ==9.12.4 ghc ==9.14.1
+tested-with:
+    ghc ==9.6.7 ghc ==9.8.4 ghc ==9.10.3 ghc ==9.12.4 ghc ==9.14.1
+
 homepage:        https://github.com/mrkkrp/megaparsec
 bug-reports:     https://github.com/mrkkrp/megaparsec/issues
 synopsis:        Monadic parser combinators
@@ -58,7 +60,7 @@ library
     default-language: Haskell2010
     build-depends:
         array >=0.5.3 && <0.6,
-        base >=4.20 && <5,
+        base >=4.18 && <5,
         bytestring >=0.2 && <0.13,
         case-insensitive >=1.2 && <1.3,
         containers >=0.5 && <0.9,
@@ -86,7 +88,7 @@ benchmark bench-speed
     hs-source-dirs:   bench/speed
     default-language: Haskell2010
     build-depends:
-        base >=4.20 && <5,
+        base >=4.18 && <5,
         bytestring >=0.2 && <0.13,
         containers >=0.5 && <0.9,
         criterion >=0.6.2.1 && <1.7,
@@ -108,7 +110,7 @@ benchmark bench-memory
     hs-source-dirs:   bench/memory
     default-language: Haskell2010
     build-depends:
-        base >=4.20 && <5,
+        base >=4.18 && <5,
         bytestring >=0.2 && <0.13,
         containers >=0.5 && <0.9,
         deepseq >=1.3 && <1.6,

